File: engine/language-server/src/main/scala/org/enso/languageserver/boot/resource/InitializationComponent.scala
package org.enso.languageserver.boot.resource

import scala.concurrent.Future

/** A component that should be initialized. */
trait InitializationComponent {

  def isInitialized: Boolean

  /** Initialize the component. */
  def init(): Future[InitializationComponent.Initialized.type]
}

object InitializationComponent {

  case object Initialized
}

File: engine/language-server/src/main/java/org/enso/languageserver/boot/resource/InitializationComponent.java
package org.enso.languageserver.boot.resource;

import java.util.concurrent.CompletableFuture;

/** A component that should be initialized. */
public interface InitializationComponent {

  /** @return `true` if the component is initialized */
  boolean isInitialized();

  /** Initialize the component. */
  CompletableFuture<Void> init();
}
